### ISSUE TITLE
New TensorRT release introduces NaNs in 3D Resnext on 16XX GPUs

### ISSUE THREAD

**Issue by hyperfraise** [NVIDIA/TensorRT#1408]
## Description

With latest TensorRT docker image, some 3d models can't be used with GPUs 1650 up to 1660 Ti with FP16 precision anymore.

## Environment

**TensorRT Version**:  7.0.0.1 -> 7.2.2.3
**NVIDIA GPU**: 1650 to 1660 Ti
**NVIDIA Driver Version**: 465
**Operating System**: Ubuntu
**PyTorch Version (if applicable)**: 1.9.0

## Steps To Reproduce

I'm comparing both those images from NVCR : `nvcr.io/nvidia/tensorrt:21.06-py3` and `nvcr.io/nvidia/tensorrt:20.03-py3`

```
nvidia-docker run -it nvcr.io/nvidia/tensorrt:2X.0X-py3 /bin/bash
```

Inside the container, get the repository for the model : 

```
git clone https://github.com/kenshohara/3D-ResNets-PyTorch.git
cd 3D-ResNets-PyTorch 
git checkout 35640f358c05904ab53816d8da0f2d968b4b3038 # current master is broken
pip install torch torchvision sklearn
```

Export a simple resnet model

```
python3
import torch
import torchvision
from models import resnext

dummy_input = torch.randn(1, 3, 35, 224, 224).float().cuda()
model = resnext.resnet50(sample_size=224, sample_duration=35).eval().cuda()

with torch.no_grad():
    torch.onnx.export(
        model,
        dummy_input,
        "resnet.onnx",
        verbose=True,
    )

```

Then 
```
/usr/src/tensorrt/bin/trtexec --onnx=resnet.onnx --fp16 --workspace=500 --saveEngine=resnet.trt --inputIOFormats=fp32:chw --outputIOFormats=fp32:chw
````

Finally, do a simple forward pass to get output values : 
```
python3
import tensorrt as trt
import pycuda.driver as cuda
import pycuda.autoinit
import os
import sys
import numpy as np

class HostDeviceMem(object):
    def __init__(self, host_mem, device_mem):
        self.host = host_mem
        self.device = device_mem
    def __str__(self):
        return "Host:\n" + str(self.host) + "\nDevice:\n" + str(self.device)
    def __repr__(self):
        return self.__str__()



def allocate_buffers(engine):
    inputs = []
    outputs = []
    bindings = []
    stream = cuda.Stream()
    for binding in engine:
        size = trt.volume(engine.get_binding_shape(binding)) * engine.max_batch_size
        dtype = trt.nptype(engine.get_binding_dtype(binding))
        # Allocate host and device buffers
        host_mem = cuda.pagelocked_empty(size, dtype)
        device_mem = cuda.mem_alloc(host_mem.nbytes)
        # Append the device buffer to device bindings.
        bindings.append(int(device_mem))
        # Append to the appropriate list.
        if engine.binding_is_input(binding):
            inputs.append(HostDeviceMem(host_mem, device_mem))
        else:
            outputs.append(HostDeviceMem(host_mem, device_mem))
    return inputs, outputs, bindings, stream



def get_engine(engine_file_path=""):
    """Attempts to load a serialized engine if available, otherwise builds a new TensorRT engine and saves it."""
    assert os.path.exists(engine_file_path), "Engine file doesn't exist"
    # If a serialized engine exists, use it instead of building an engine.
    print("Reading engine from file {}".format(engine_file_path))
    with open(engine_file_path, "rb") as f, trt.Runtime(TRT_LOGGER) as runtime:
        return runtime.deserialize_cuda_engine(f.read())


TRT_LOGGER = trt.Logger()

engine_file_path="resnet.trt"
with get_engine(
    engine_file_path
) as engine, engine.create_execution_context() as context:
    inputs, outputs, bindings, stream = allocate_buffers(engine)
    # Do inference
    # Set host input to the image. The do_inference function will copy the input to the GPU before executing.
    np.random.seed(17)
    inputs[0].host = np.array(
        np.random.choice(range(256), (1, 3, 35, 224, 224)),
        dtype=np.float32,
    )
    [cuda.memcpy_htod_async(inp.device, inp.host, stream) for inp in inputs]
    context.execute_async_v2(bindings=bindings, stream_handle=stream.handle)
    [
        cuda.memcpy_dtoh_async(out.host, out.device, stream)
        for out in outputs
    ]
    stream.synchronize()
    print([out.host for out in outputs])

```

Using this with 21.06 gives nans, and normal floats with 20.03.

Interestingly, reducing the duration to 16 and the image size to 112 fixes the problem.

**Comment by ttyio** [comment]
Hello @hyperfraise , have you tried 8.0GA ? is this fixed? thanks!

**Comment by hyperfraise** [comment]
@ttyio  still NaNs with 21.08, which is TensorRT 8.0.1.6

**Comment by ttyio** [comment]
@hyperfraise sounds TRT bug, and I have create internal bug to track this, sorry...

**Comment by hyperfraise** [comment]
**Thank you**

**Comment by hyperfraise** [comment]
Still nans with 21.09, which is TensorRT 8.0.3

**Comment by hyperfraise** [comment]
Problem still exists with 21.11

**Comment by hyperfraise** [comment]
Great news (for me). Problem is fixed in 21.12, that is TensorRT 8.2 @ttyio
